### PR TITLE
Added accounts and login features

### LINKED ISSUE #18: Registration Page
# Features to Include

The page should contain inputs for the following fields (check ER Diagram for data model constraints):

- [ ] First Name
- [ ] Last Name (optional)
- [ ] Email Address (ensure this is a personal email address)
- [ ] Password (with password strength check)
- [ ] Password Confirmation (with frontend/backend verification of matching fields)
- [ ] Phone Number (optional)
- [ ] Gender 

## Notes

- Would have loved to setup email verification by sending an email to the user, but CCC deployment won't have the SMTP port (587) open by default. Would need to take additional permission to enable mailers.
- Link to login page from this page.
- Set username as email address on backend.

### PR DESCRIPTION
**Pull request by anirudhprabhakaran3**
# Description

Adds accounts and login to Corpus.

Fixes #18, #19

## Dependencies

None

## Type of Change

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration. (for bug fix / breaking change)
_Put an `x` in the boxes that apply_

- [ ] Test A
- [ ] Test B

---

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by anirudhprabhakaran3** [comment]
Oh I have to add tests for accounts. Will do that over this week and weekend.

**Comment by nishant-nayak** [comment]
Convert your PR to draft if tests are pending to be added.

File: corpus/templates/accounts/login.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{% extends 'base.html' %}
+
+{% block title %}
+Login
+{% endblock %}
+
+{% block content %}
+<div class="card w-1/2 mx-auto my-10 bg-base-100 glass shadow-xl prose">
```
Comment (nishant-nayak) [inline corpus/templates/accounts/login.html:8]: Not really a fan of the glass theme here as well, I think you can take reference from popular website login pages (LeetCode is a good example) to style the page better.

**Review by nishant-nayak — CHANGES_REQUESTED** [review]
Left a bunch of comments, mostly styling related.

Logic wise code seems alright, if you're stuck on styling let me know and I'll try pitching in some code.

File: corpus/accounts/tests.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            "email": "testuser@gmail.com",
+            "phone_no": "0000000000",
+            "gender": "M",
+            "password1": "user1@password",
+            "password2": "user1@password",
+        }
+
+        response = self.client.post(self.signup_path, form_data, follow=True)
+
+        self.assertEqual(response.status_code, 200)
```
[the remark this answers is not in the document]
Comment (anirudhprabhakaran3) [reply corpus/accounts/tests.py:60]: Apparently this is what Django returns. Even the server logs return a 200.

https://stackoverflow.com/questions/25839434/django-login-with-wrong-credentials-returns-200-not-401

File: corpus/templates/accounts/signup.html
[the remark this answers is not in the document]
Comment (anirudhprabhakaran3) [reply corpus/templates/accounts/signup.html:1]: Regarding the phone number thing - does that like apply for international numbers also? That's why I didn't put anything there for now 😅

File: corpus/accounts/tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nishant-nayak) [reply corpus/accounts/tests.py:60]: Interesting. I suppose this is fine as long as this is the standard we maintain.

[the remark this answers is not in the document]
Comment (nishant-nayak) [reply corpus/accounts/tests.py:60]: Interesting. I suppose this is fine as long as this is the standard we maintain.

File: corpus/templates/accounts/signup.html
[the remark this answers is not in the document]
Comment (nishant-nayak) [reply corpus/templates/accounts/signup.html:1]: I suppose we can have some other kind of validation in place for phone numbers. I just learnt the shortest phone numbers can be like 7-8 digits xD

**Review by nishant-nayak — APPROVED** [review]
LGTM!

**Pull request merged**